### PR TITLE
Correct typeof result for one of the Union types

### PR DESCRIPTION
**Pull request by pdbartlett**

### CODE REVIEW

File: doc/src/manual/types.md [hunk 1]
```diff
@@ -1153,7 +1153,7 @@ julia> typeof(Rational{Int})
 DataType
 
 julia> typeof(Union{Real,Float64,Rational})
-DataType
+Union
```
Comment (fredrikekre) [inline doc/src/manual/types.md]: I get `DataType` though?
```
julia> typeof(Union{Real,Float64,Rational})
DataType
```

**Comment by KristofferC** [comment]
This output is tested as part of CI so it is strange that it would be wrong.

**Comment by yuyichao** [comment]
`Union{Real,Float64,Rational}` is `Real` so the doc is correct. It seems that this didn't used to be the case in the 0.1/0.2 time frame. Ref https://github.com/JuliaLang/julia/commit/403b0d1e7b164236301d28f5684622bf39b996e8#diff-546beeead6861ec5447c9a4e3efe8bd8R1030 https://github.com/JuliaLang/julia/commit/1d1a235319eb185f833683df3cdc1a009d3f9c43#diff-546beeead6861ec5447c9a4e3efe8bd8R992 and https://github.com/JuliaLang/julia/commit/b995da566412b3a8f56fce282f4324e42c425577#diff-546beeead6861ec5447c9a4e3efe8bd8R1083 . This one should probably just be removed since the union type is shown below.

**Review by yuyichao — CHANGES_REQUESTED** [review]
Remove this union case.

File: doc/src/manual/types.md (hunk 1 above)
Comment (pdbartlett) [reply doc/src/manual/types.md]: Strange, so do I now. Maybe I didn't try it exactly as written :(

Anyhow, removed entirely at @yuyichao's suggestion.

**Review by yuyichao — APPROVED** [review]
Please don't `at` people in commit message, or it'll send a message whenever that commit is rebased.

LGTM. I glanced over the doc on this page and didn't see any explicit mention of `Union` "merging" types so maybe that could be added later. I don't think the code removed here counts as a document of that so that change shouldn't be part of this PR.

**Review by Keno — APPROVED** [review]

**Pull request merged**